spec/services/preserved_object_handler_update_version_spec.rb

```diff
@@ -377,62 +438,99 @@
+              end
             end
           end
-          context 'unchanged' do
-            it 'version' do
-              orig = pc.version
-              po_handler.update_version_after_validation
-              expect(pc.reload.version).to eq orig
+        end
+        context 'checksums_validated = true' do
+          context 'PreservedCopy' do
+            it 'last_moab_validation updated' do
+              expect do
+                po_handler.update_version_after_validation(true)
+              end.to change { pc.reload.status }
             end
-            it 'size' do
-              orig = pc.size
-              po_handler.update_version_after_validation
-              expect(pc.reload.size).to eq orig
+            it 'size updated to incoming_size' do
+              expect do
+                po_handler.update_version_after_validation(true)
+              end.to change { pc.reload.size }.to(incoming_size)
             end
-            it 'last_version_audit' do
-              orig = pc.last_version_audit
-              po_handler.update_version_after_validation
-              expect(pc.reload.last_version_audit).to eq orig
+            it 'last_version_audit updated' do
+              expect do
+                po_handler.update_version_after_validation(true)
+              end.to change { pc.reload.last_version_audit }
             end
-            context 'original status was invalid_moab' do
-              before { pc.update(status: 'invalid_moab') }
-              it 'status remains invalid_moab if caller has not verified checksums' do
-                expect do
-                  po_handler.update_version_after_validation
-                end.not_to change { pc.reload.status }.from('invalid_moab')
-              end
-              it 'status remains invalid_moab evne if caller has verified checksums' do
+            it 'version updated to incoming_version' do
+              expect do
+                po_handler.update_version_after_validation(true)
+              end.to change { pc.reload.version }.from(2).to(incoming_version)
+            end
+            context 'status' do
+              it 'starting status invalid_moab unchanged' do
+                pc.update(status: 'invalid_moab')
                 expect do
                   po_handler.update_version_after_validation(true)
                 end.not_to change { pc.reload.status }.from('invalid_moab')
               end
+              context 'starting status was not invalid_moab' do
+                shared_examples 'POH#update_version_after_validation(true) changes status to "invalid_moab"' do |orig_status|
+                  before { pc.update(status: orig_status) }
+                  it "original status #{orig_status}" do
+                    expect do
+                      po_handler.update_version_after_validation(true)
+                    end.to change { pc.reload.status }.from(orig_status).to('invalid_moab')
```

Comment:
if checksums_validated arg is true, and moab is invalid, then the status is invalid_moab because we assume all versions were checksum validated at some point;  the moab validation errors take precedence